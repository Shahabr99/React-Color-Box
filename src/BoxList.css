.container {
  width: 1000px;
  height: 1000px;
  border: 1px solid #000;
  border-radius: 15px;
  margin: 100px auto;
  box-shadow: 1px 3px 3px #00000067;
}