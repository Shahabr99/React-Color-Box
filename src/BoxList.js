import React, {useState} from 'react';
import { v4 as uuid } from 'uuid'
import NewBoxForm from './NewBoxForm';
import Box from './Box'
import './BoxList.css'

const BoxList = () => {
  const INITIAL_STATE = [
    {id: uuid(), width:"", height:"", color:""}
  ]

  const [boxes, setBoxes] = useState(INITIAL_STATE)

  const addBox = (width, height, color) => {
    setBoxes(boxes => ([
      ...boxes,
      { width, height, color, id: uuid()}
    ]))
  }

  return (
    <>
      <h2>Create Colorful Boxes</h2>
      <NewBoxForm addBox={addBox} />
      <div className="container">
        {boxes.map(box => <Box key={box.id} width={box.width} height={box.height} color={box.color} />)}
      </div>
    </>
    
  )
}

export default BoxList;