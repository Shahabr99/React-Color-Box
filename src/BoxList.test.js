import React from 'react';
import {render, screen, fireEvent, act } from "@testing-library/react";
import BoxList from './BoxList';

it("should render BoxList without failure", function() {
  render(<BoxList />)
})

it("matches snapshot", function() {
  const { asFragment } = render(<BoxList />)
  expect(asFragment()).toMatchSnapshot()
})

it("should add a new box", async function() {
  render(<BoxList />)
  const inputWidth = screen.getByPlaceholderText("Box Width");
  const inputHeight = screen.getByPlaceholderText("Box Height");
  const inputColor = screen.getByPlaceholderText("Box Color");
  const btn = screen.queryByText("Create ðŸ“¦");

  expect(screen.queryByText('red')).not.toBeInTheDocument();

  await act(async () => {
    fireEvent.change(inputColor, {target: { value: 'red'}});
    fireEvent.change(inputWidth, {target: { value: '50px'}});
    fireEvent.change(inputHeight, {target: { value: '50px'}});
    fireEvent.click(btn);
  });
  

  expect(screen.getByDisplayValue('red')).toBeInTheDocument();
})