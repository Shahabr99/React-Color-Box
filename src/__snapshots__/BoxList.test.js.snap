// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form
    action="#"
  >
    <div>
      <label
        for="width"
      >
        Width:
      </label>
      <input
        id="width"
        name="width"
        placeholder="Box Width"
        value=""
      />
    </div>
    <div>
      <label
        for="height"
      >
        Height
      </label>
      <input
        id="height"
        name="height"
        placeholder="Box Height"
        value=""
      />
    </div>
    <div>
      <label
        for="color"
      >
        Background color:
      </label>
      <input
        id="color"
        name="color"
        placeholder="Box Color"
        value=""
      />
    </div>
    <button
      type="submit"
    >
      Create ðŸ“¦
    </button>
  </form>
  <div
    class="container"
  >
    <div
      class="box"
    />
  </div>
</DocumentFragment>
`;
